SingularityService/src/main/java/com/hubspot/singularity/SingularityManagedScheduledExecutorServiceProvider.java

```diff
@@ -0,0 +1,45 @@
+package com.hubspot.singularity;
+
+import static com.google.common.base.Preconditions.checkState;
+import io.dropwizard.lifecycle.Managed;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import javax.inject.Provider;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+
+public class SingularityManagedScheduledExecutorServiceProvider implements Provider<ScheduledExecutorService>, Managed {
+
+  private final AtomicBoolean stopped = new AtomicBoolean();
+  private ScheduledExecutorService service;
+
+  private final long timeoutInSeconds;
+
+  public SingularityManagedScheduledExecutorServiceProvider(final int poolSize, final long timeoutInSeconds, final String name) {
+    this.service = Executors.newScheduledThreadPool(poolSize, new ThreadFactoryBuilder().setNameFormat(name + "-pool-%d").setDaemon(true).build());
+    this.timeoutInSeconds = timeoutInSeconds;
+  }
+
+  @Override
+  public synchronized ScheduledExecutorService get() {
```

Comment:
You can probably drop the `synchronized` lock here given you are reading from a thread safe data structure. Also there's no corresponding lock on the writer via the `stop` method below.